From 4085ab7491b402bc675a42580f844825686c43a4 Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Fri, 17 Feb 2017 13:33:30 +0000
Subject: [PATCH] Remove the Printers panel

Continuous does not ship Samba, so we need to patch the Printers panel
out, given that Control Center does not allow disabling it at build
time.
---
 configure.ac            | 37 -------------------------------------
 panels/Makefile.am      |  3 +--
 shell/Makefile.am       |  3 +--
 shell/cc-panel-loader.c |  2 --
 4 files changed, 2 insertions(+), 43 deletions(-)

diff --git a/configure.ac b/configure.ac
index 12a8ade..4e02317 100644
--- a/configure.ac
+++ b/configure.ac
@@ -147,9 +147,6 @@ PKG_CHECK_MODULES(COLOR_PANEL, $COMMON_MODULES
                   colord-gtk >= $COLORD_GTK_REQUIRED_VERSION
                   libsoup-2.4
                   gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION)
-PKG_CHECK_MODULES(PRINTERS_PANEL, $COMMON_MODULES
-                  polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
-                  smbclient)
 PKG_CHECK_MODULES(PRIVACY_PANEL, $COMMON_MODULES)
 PKG_CHECK_MODULES(REGION_PANEL, $COMMON_MODULES
                   polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
@@ -266,40 +263,8 @@ if test x${have_bluetooth} = xyes; then
   AC_DEFINE(HAVE_BLUETOOTH, 1, [Define to 1 if bluetooth support is available])
 fi
 
-# Check for CUPS 1.4 or newer
 AC_PROG_SED
 
-AC_PATH_PROG(CUPS_CONFIG, cups-config)
-
-if test x$CUPS_CONFIG = x; then
-  AC_MSG_ERROR([cups-config not found])
-fi
-
-CUPS_API_VERSION=`$CUPS_CONFIG --api-version`
-CUPS_API_MAJOR=`echo $ECHO_N $CUPS_API_VERSION | cut -d . -f 1`
-CUPS_API_MINOR=`echo $ECHO_N $CUPS_API_VERSION | cut -d . -f 2`
-
-AC_CHECK_HEADERS([cups/cups.h cups/http.h cups/ipp.h cups/ppd.h],,
-                 AC_MSG_ERROR([CUPS headers not found]))
-
-if ! test $CUPS_API_MAJOR -gt 1 -o \
-          $CUPS_API_MAJOR -eq 1 -a $CUPS_API_MINOR -ge 4 ; then
-  AC_MSG_ERROR([CUPS 1.4 or newer not found])
-fi
-
-# https://bugzilla.gnome.org/show_bug.cgi?id=696766
-CUPS_CPPFLAGS=""
-if test $CUPS_API_MAJOR -gt 1 -o \
-        $CUPS_API_MAJOR -eq 1 -a $CUPS_API_MINOR -ge 6 ; then
-   CUPS_CPPFLAGS=-D_PPD_DEPRECATED=""
-fi
-
-CUPS_CFLAGS=`$CUPS_CONFIG --cflags | $SED -e 's/-O\w*//g' -e 's/-m\w*//g'`
-CUPS_LIBS=`$CUPS_CONFIG --libs`
-AC_SUBST(CUPS_CPPFLAGS)
-AC_SUBST(CUPS_CFLAGS)
-AC_SUBST(CUPS_LIBS)
-
 # Optional dependency for the user accounts panel
 AC_ARG_WITH([cheese],
             AS_HELP_STRING([--with-cheese], [enable cheese webcam support]),,
@@ -534,8 +499,6 @@ panels/color/icons/48x48/Makefile
 panels/color/icons/64x64/Makefile
 panels/color/icons/256x256/Makefile
 panels/color/icons/scalable/Makefile
-panels/printers/Makefile
-panels/printers/gnome-printers-panel.desktop.in
 panels/privacy/Makefile
 panels/privacy/gnome-privacy-panel.desktop.in
 panels/network/Makefile
diff --git a/panels/Makefile.am b/panels/Makefile.am
index 507cd2f..5e4c51c 100644
--- a/panels/Makefile.am
+++ b/panels/Makefile.am
@@ -16,8 +16,7 @@ SUBDIRS= \
 	datetime \
 	search \
 	privacy \
-	sharing \
-	printers
+	sharing
 
 if BUILD_WACOM
 SUBDIRS += wacom
diff --git a/shell/Makefile.am b/shell/Makefile.am
index 6af3199..04a0491 100644
--- a/shell/Makefile.am
+++ b/shell/Makefile.am
@@ -93,8 +93,7 @@ gnome_control_center_LDADD =						\
 	$(top_builddir)/panels/sharing/libsharing.la			\
 	$(top_builddir)/panels/sound/libsound.la			\
 	$(top_builddir)/panels/universal-access/libuniversal-access.la	\
-	$(top_builddir)/panels/user-accounts/libuser-accounts.la	\
-	$(top_builddir)/panels/printers/libprinters.la
+	$(top_builddir)/panels/user-accounts/libuser-accounts.la
 
 if BUILD_WACOM
 gnome_control_center_LDADD += $(top_builddir)/panels/wacom/libwacom-properties.la
diff --git a/shell/cc-panel-loader.c b/shell/cc-panel-loader.c
index 20ebb85..cc00466 100644
--- a/shell/cc-panel-loader.c
+++ b/shell/cc-panel-loader.c
@@ -45,7 +45,6 @@ extern GType cc_network_panel_get_type (void);
 extern GType cc_notifications_panel_get_type (void);
 extern GType cc_goa_panel_get_type (void);
 extern GType cc_power_panel_get_type (void);
-extern GType cc_printers_panel_get_type (void);
 extern GType cc_privacy_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
@@ -87,7 +86,6 @@ static struct {
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type),
   PANEL_TYPE("online-accounts",  cc_goa_panel_get_type          ),
   PANEL_TYPE("power",            cc_power_panel_get_type        ),
-  PANEL_TYPE("printers",         cc_printers_panel_get_type     ),
   PANEL_TYPE("privacy",          cc_privacy_panel_get_type      ),
   PANEL_TYPE("region",           cc_region_panel_get_type       ),
   PANEL_TYPE("search",           cc_search_panel_get_type       ),
-- 
2.9.3

